spreadsheets.js

```diff
@@ -422,8 +438,103 @@ function update_schedule(service_account_auth, key, colname, format, schedule, c
     });
 }
 
+// parse the input string for a results update
+function parse_result(input_string){
+    var tokens = get_tokens(input_string);   
+    var result = find_result(tokens);
+    var players = find_players(tokens);
+    
+    return {
+        "white": players.white,
+        "black": players.black,
+        "result": result,
+    };
+}
+
+function get_tokens(input_string){
+    return input_string.split(" ");
+}
+
+function find_result(tokens){
+    var result;
+    _.some(tokens, function(token){
+        return (result = VALID_RESULTS[token.toUpperCase()]) ? true : false;
+    });
+    return result;
+}
+
+function find_players(tokens){
+    // in reality, the order is arbitrary.
+    // just assuming first I find is white, the second is black
+    var players = {};
+    
+    var player_tokens = filter_player_tokens(tokens);
+
+    //assuming we found 2 tokens, we should convert them to player names
+    if(player_tokens.length == 2){
+        players.white = player_tokens[0];
+        players.black = player_tokens[1];
+    }
+
+    return players;
+}
+
+function filter_player_tokens(tokens){
+    return _.filter(tokens, function(token){
+        return /^\<@[A-Z0-9]+\>$/.test(token);
+    });
+}
+
+//this assumes it is being given a valid result - that is th contract
+//error handling for a bad result should happen outside this function
+function update_result(service_account_auth, key, colname, result, callback){
+    var white = result.white;
+    var black = result.black;
+    var result = result.result;
+    
+    find_pairing(
+        service_account_auth, 
+        key, 
+        white.name, 
+        black.name, 
+        function(err, row, reversed){
+            if(err){
+                return callback(err);
+            }
+            var result_cell = row[colname];
+            console.log(result_cell);
```

Comment:
Did you intentionally leave this in? I am ok with us leaving in a certain amount of logging, and think it would be worth using the various console.* methods, like console.error for actual errors, console.log for just logging info etc.